### PR TITLE
Added start/stop capabilities through http

### PR DESCRIPTION
**Pull request by jdurnwald**
Small enhancement to add express to enable a start/stop mechanism for the simulators through a small web page.

### CODE REVIEW

**Review by jerome-benoit — APPROVED** [review]
- I think you can document a bit the PORT environment variable usage in the HTTP server; 
- Please have a look at CI run failures, you seem to have not made a `npm install` from the npm version shipped with the current node.js LTS.

Thanks.

**Comment by jerome-benoit** [comment]
@jdurnwald: I think we need to discuss the overall goals you want to reach here: 

+ Once deployed on CF, stopping or starting the simulator is a matter of starting or stopping the app; 
+ Doing actions on a running simulated CS need to use the UI service already implemented. 

The purpose of the current UI is to avoid CLI usage for local simulator usage?

**Comment by jerome-benoit** [comment]
> The simulator is being used to assist in demonstrations of the E-Mobility solution to customers by any number of our presales community. The challenge is that they may not have access to the admin level of the app in BTP in order to stop and start it. So, the question is why they would need to do that? There is a root problem that hasn't been resolved in that the simulator will stop pushing data after a period of time (unsure of the amount the time it takes) when running in BTP 

The automatic transaction generator (ATG) is configurable at charging station template level:  

```json
"AutomaticTransactionGenerator": {
    "enable": false,
    "minDuration": 60,
    "maxDuration": 80,
    "minDelayBetweenTwoTransactions": 15,
    "maxDelayBetweenTwoTransactions": 30,
    "probabilityOfStart": 1,
    "stopAfterHours": 0.3,
    "stopOnConnectionFailure": true,
    "requireAuthorize": true
  }
```

If the issue is that the `stopAfterHours` is reached, then it needs to be configured to 3650 for 10 years, or something like an unlimited ATG run feature need to be added. 
If the issue is something else, it needs to be reported here as an issue to be analysed and fixed. 

> and the only current manner to mitigate that is to restart the app itself. This is being done manually by our demo team several times a day. If this were to be resolved, then a restart mechanism would not be necessary.

The workaround is acceptable if the main issue is too difficult to fix. But an exhaustive bug report need to done first, so the project have a chance to fix the root cause. 

Thanks.

**Comment by jdurnwald** [comment]
> > The simulator is being used to assist in demonstrations of the E-Mobility solution to customers by any number of our presales community. The challenge is that they may not have access to the admin level of the app in BTP in order to stop and start it. So, the question is why they would need to do that? There is a root problem that hasn't been resolved in that the simulator will stop pushing data after a period of time (unsure of the amount the time it takes) when running in BTP
> 
> The automatic transaction generator (ATG) is configurable at charging station template level:
> 
> ```json
> "AutomaticTransactionGenerator": {
>     "enable": false,
>     "minDuration": 60,
>     "maxDuration": 80,
>     "minDelayBetweenTwoTransactions": 15,
>     "maxDelayBetweenTwoTransactions": 30,
>     "probabilityOfStart": 1,
>     "stopAfterHours": 0.3,
>     "stopOnConnectionFailure": true,
>     "requireAuthorize": true
>   }
> ```
> 
> If the issue is that the `stopAfterHours` is reached, then it needs to be configured to 3650x24 for 10 years, or something like an unlimited ATG run feature need to be added. If the issue is something else, it needs to be reported here as an issue to be analysed and fixed.
> 
> > and the only current manner to mitigate that is to restart the app itself. This is being done manually by our demo team several times a day. If this were to be resolved, then a restart mechanism would not be necessary.
> 
> The workaround is acceptable if the main issue is too difficult to fix. But an exhaustive bug report need to be done first, so the project have a chance to fix the root cause.
> 
> Thanks.

After talking again with the team, when demonstrating the solution, the presenter will sometimes want to stop the simulator to mimic unexpected downtime (blackouts) and how the solution responds to that.  So, again, the presenter will not have admin access to BTP to do that and this is the point of the manual restart capability.

Hope this helps to clarify.

Jeff.

**Comment by jerome-benoit** [comment]
> 
> After talking again with the team, when demonstrating the solution, the presenter will sometimes want to stop the simulator to mimic unexpected downtime (blackouts) and how the solution responds to that. So, again, the presenter will not have admin access to BTP to do that and this is the point of the manual restart capability.
> 
> Hope this helps to clarify.

That PR can merged even if to reach that goal, the proper way to do it is to use the UI service under development. 
But for the merge to happen, you have to: 
+ address review comments; 
+ fix merge conflicts; 
+ fix GitHub CI actions issues.

File: src/http/start.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+app.get('/stop', (request, response, next) => {
+  Bootstrap.getInstance().stop().catch(next);
+  console.log('*** stopped');
+  response.send('<b>Stopped</b><br/><a href="/">Return to Top</a>');
+});
+
+app.listen(process.env.PORT ?? 8080, () =>
+  console.log(`Listening on port: ${process.env.PORT ?? 8080}`)
+);
```
Comment (jerome-benoit) [inline src/http/start.ts:32]: I think the correct path for a UI is under an independent path like ui/httpd with its own package.json, etc. 
For now, could you please rename the path to src/ui/httpd?

Comment (jerome-benoit) [reply src/http/start.ts:32]: You just push the changes under the branch under your cloned repo referenced in that PR.

Comment (jerome-benoit) [reply src/http/start.ts]: You  still need to: 

+ fix GitHub CI actions issues.

Comment (jdurnwald) [reply src/http/start.ts]: I upgraded node.js to v16.14.2 and did the npm install like you suggested.  I committed the package-lock.json, but not sure if that should be done.

Thanks for your patience.

**Comment by jdurnwald** [comment]
Thanks for the merge.  Are the CI failures minimal?

**Pull request merged**